bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/service.js

```diff
@@ -0,0 +1,121 @@
+import Presentations from '/imports/api/2.0/presentations';
+import Auth from '/imports/ui/services/auth';
+import { makeCall } from '/imports/ui/services/api';
+
+const CONVERSION_TIMEOUT = 300000;
+
+// fetch doens't support progress. So we use xhr which support progress.
+const futch = (url, opts = {}, onProgress) => new Promise((res, rej) => {
+  const xhr = new XMLHttpRequest();
+
+  xhr.open(opts.method || 'get', url);
+
+  Object.keys(opts.headers || {})
+    .forEach(k => xhr.setRequestHeader(k, opts.headers[k]));
+
+  xhr.onload = (e) => {
+    if (e.target.status !== 200) {
+      return rej({ code: e.target.status, message: e.target.statusText });
+    }
+
+    return res(e.target.responseText);
+  };
+  xhr.onerror = rej;
+  if (xhr.upload && onProgress) {
+    xhr.upload.addEventListener('progress', onProgress, false);
+  }
+  xhr.send(opts.body);
+});
+
+const getPresentations = () =>
+  Presentations
+    .find()
+    .fetch()
+    .map(presentation => ({
+      id: presentation.id,
+      filename: presentation.name,
+      isCurrent: presentation.current || false,
+      upload: { done: true, error: false },
+      conversion: presentation.conversion || { done: true, error: false },
+    }));
+
+const observePresentationConversion = (meetingId, filename) => new Promise((resolve, reject) => {
+  const conversionTimeout = setTimeout(() => {
+    reject({
+      filename,
+      message: 'Conversion timeout.',
+    });
+  }, CONVERSION_TIMEOUT);
+
+  const didValidate = () => {
+    clearTimeout(conversionTimeout);
+    resolve(filename);
+  };
+
+  Tracker.autorun((c) => {
+    const query = Presentations.find({ meetingId, filename });
+
+    query.observeChanges({
+      changed: (id, fields) => {
+        if (fields.conversion.done) {
+          c.stop();
+          didValidate();
+        }
+      },
+    });
+  });
+});
+
+const uploadPresentation = (file, meetingID, endpoint, onError, onProgress) => {
+  const data = new FormData();
+  data.append('presentation_name', file.filename);
+  data.append('Filename', file.filename);
```

Comment:
We have such a mix of different naming conventions here
camelCase - `'fileUpload'`
underscore - `'presentation_name'` , `'is_downloadable'`
not sure what type this is - `'Filename'`

Probably this needs to be corrected.